corehq/apps/linked_domain/updates.py

```diff
@@ -347,6 +347,8 @@ def update_fixture(domain_link, tag, overwrite=False):
     clear_fixture_cache(domain_link.linked_domain)
 
 
+#TODO: Limit the scope of this atomicity. Ideally, we fetch and prep the data prior to entering the atomic block
+@transaction.atomic
```

Comment:
Could that be done using `with transaction.atomic():` around the limited code that should be atomic inside this function?

Reply:
I left the note here, but I don't feel super comfortable about trying to fix it as part of this work. I could certainly move the transaction around the role loops, but I don't think that changes the scope of the transaction enough to warrant the change.